src/main/webapp/user_profile_script.js

```diff
@@ -95,47 +106,39 @@ function showOfferHelp() {
 }
 
 function showModal() {
-    var modal = document.getElementById("createTaskModal");
+    var modal = document.getElementById("createTaskModalWrapper");
     modal.style.display = "block";
 }
 
 function closeModal() {
-    var modal = document.getElementById("createTaskModal");
+    var modal = document.getElementById("createTaskModalWrapper");
     modal.style.display = "none";
 }
 
 function closeEditModal() {
-    var modal = document.getElementById("editTaskModal");
+    var modal = document.getElementById("editTaskModalWrapper");
     modal.style.display = "none";
 }
 
 // If the user clicks outside of the modal, closes the modal directly
 window.onclick = function(event) {
-    var modal = document.getElementById("createTaskModal");
+    var modal = document.getElementById("createTaskModalWrapper");
     if (event.target == modal) {
         modal.style.display = "none";
     }
-    var editModal = document.getElementById("editTaskModal");
+    var editModal = document.getElementById("editTaskModalWrapper");
     if (event.target == editModal) {
         editModal.style.display = "none";
     }
 }
 
 async function displayNeedHelpTasks() {
-    const queryURL = "mytasks?keyword=Owner";
+    const queryURL = "/mytasks?keyword=Owner";
     const request = new Request(queryURL, {method: "GET"});
     const response = await fetch(request);
     const taskResponse = await response.json();
     const needHelpTable = document.getElementById("need-help");
-    needHelpTable.innerHTML = "";
-    var headerRow = document.createElement("tr");
-    var headers = ["Task Overview", "Helper", "Status", "Edit", "Delete"];
-    for (var index = 0; index < headers.length; index++) {
-        var th = document.createElement("th");
-        th.appendChild(document.createTextNode(headers[index]));
-        headerRow.appendChild(th);
-    }
-    needHelpTable.appendChild(headerRow);
+    needHelpTable.innerHTML = needHelpTable.rows[0].innerHTML;
```

Comment:
I don't think this is doing quite what we want, semantically. Here's the HTML I see on my local instance:

```
<table class="task-table" id="need-help" style="display: table;">
    <tbody>
        <tr>
            <th>Task Overview</th>
            <th>Helper</th>
            <th>Status</th>
            <th>Edit</th>
            <th>Delete</th>
        </tr>
    </tbody>
    <tr>
        <td>this is my task</td>
        <td>N/A</td>
        <td>OPEN</td>
        <td><button class="edit-task"><i class="fa fa-edit"></i></button></td><td><button class="delete-task"><i class="fa fa-trash-o"></i></button></td>
    </tr>
</table>
```

See how the table has a `tbody` child with the header rows, then a `tr` element afterwards? Better would be to wrap the header row in a `thead` element that is a child of the `table` element, then have this JS stick to modifying just the `tbody`, which should contain all the Task data rows.

Reply:
Ah using `thead` and `tbody` make things much easier. I put the header row in a `thead` element in the new commit and empty the `tbody` each time I load the table.